lib/elixir/src/elixir_erl.erl

```diff
@@ -243,13 +247,29 @@ is_macro(defmacro)  -> true;
 is_macro(defmacrop) -> true;
 is_macro(_)         -> false.
 
+usort_defs_with_meta(Entries) ->
+  % Sort and deduplicate definitions. For duplicated definitions
+  % we take the one with earliest line.
+
+  LineComparator = fun
+    ({Def1, Meta1}, {Def1, Meta2}) -> ?line(Meta1) =< ?line(Meta2);
+    ({Def1, _Meta1}, {Def2, _Meta2}) -> Def1 =< Def2
+  end,
+
+  UniqFun = fun({Def, _Meta}) -> Def end,
+
+  lists:uniq(UniqFun, lists:sort(LineComparator, Entries)).
```

Comment:
I don't think we actually need uniqueness here. We only need sorting but `lists:usort` is faster than `lists:sort`, which is why we called `usort` before.  I don't think we actually expect duplicate tuples in there!

Reply:
There can be duplicates for callbacks with multiple clauses. Looks like that's not the case for definitions though!

Reply:
Updated!